relayer/morse/context.go

```diff
@@ -0,0 +1,112 @@
+package morse
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pokt-foundation/pocket-go/provider"
+	sdkrelayer "github.com/pokt-foundation/pocket-go/relayer"
+
+	"github.com/buildwithgrove/path/relayer"
+)
+
+// requestContext provides all the functionality required by the relayer package
+// for handling a single service request.
+var _ relayer.ProtocolRequestContext = &requestContext{}
+
+// requestContext captures all the data required for handling a single service request.
+type requestContext struct {
+	fullNode  FullNode
+	endpoints map[relayer.EndpointAddr]endpoint
+	serviceID relayer.ServiceID
+
+	// selectedEndpoint is the endpoint that has been selected for sending a relay.
+	// Sending a relay will fail if this field is not set through a call to the SelectEndpoint method.
+	selectedEndpoint *endpoint
+}
+
+func (rc *requestContext) HandleServiceRequest(payload relayer.Payload) (relayer.Response, error) {
+	if rc.selectedEndpoint == nil {
+		return relayer.Response{}, fmt.Errorf("handleServiceRequest: no endpoint has been selected on service %s", rc.serviceID)
+	}
+	endpoint := rc.selectedEndpoint
```

Comment:
Can we avoid this extra assignment? I think referencing `rc.selectedEndpoitn` will be easier to understand.